src/backend/libpq/rangerrest.c

```diff
@@ -453,23 +453,30 @@ static int call_ranger_rest(CURL_HANDLE curl_handle, const char* request)
 		{
 			if (retry > 1)
 			{
-				elog(WARNING, "ranger plugin service from http://%s:%d/rps is unavailable : %s, try another http://%s:%d/rps\n",
+				/* Don't expose this warning message to client, just record in log.
+				 * The value of whereToSendOutput is DestRemote, so set it to DestNone
+				 * and set back after write a warning message in log file.
+				 */
+				CommandDest commandDest = whereToSendOutput;
+				whereToSendOutput = DestNone;
+				elog(WARNING, "ranger plugin service from http://%s:%d/rps is unavailable : %s, "
+						"trying ranger plugin service at http://%s:%d/rps\n",
```

Comment:
If so, why not just remove the code and add comments here? Does elog(LOG, ...) possibly expose this to users?